src/models/product.model.ts

```diff
@@ -0,0 +1,62 @@
+import { Column, Entity } from 'typeorm'
+import { BaseEntity } from './'
+
+@Entity({ name: 'products' })
+export abstract class Product extends BaseEntity {
+  // Attributes
+  @Column({
+    comment: 'product name',
+    length: 99,
+    type: 'varchar',
+    nullable: false
+  })
+  name: string
+
+  @Column({
+    comment: 'product description',
+    type: 'text',
+    nullable: false
+  })
+  description: string
+
+  @Column({
+    comment: 'product price',
+    type: 'decimal',
+    nullable: false
+  })
+  unit_price: number
+
+  @Column({
+    comment: 'product rating',
+    type: 'decimal',
+    default: 0.0,
+    nullable: true
+  })
+  rating: number
+
+  @Column({
+    comment: 'product visibility',
+    type: 'boolean',
+    default: false,
+    nullable: true
+  })
+  visible: boolean
+
+  @Column({
+    comment: 'product type',
+    type: 'varchar',
+    nullable: false
+  })
+  type: string
+
+  @Column({
+    comment: 'product image url',
+    type: 'varchar',
+    nullable: false
+  })
+  image_url: string
```

Comment:
When using TypeORM, you can define class property names using `camelCase` convention, while customizing the corresponding database column names in the `@Column` decorator. This allows flexibility in how your property is represented in the database. For example, if you want a property named `imageUrl` in your class but `image_url` as the column name in the database, you can do the following:

```ts
@Column({
  comment: 'product image url',
  name: 'image_url',
  type: 'varchar',
  nullable: false
})
imageUrl: string;
```

In this example, the class property `imageUrl` is mapped to the `image_url` column in the database, making the code more readable while adhering to database naming conventions.

This also applies for `unit_price`.

Reply:
Thanks, yeah sorry I did it on auto-mode when writing it forgot we are using `JS` for a moment.